src/KeyringController.ts

```diff
@@ -273,26 +325,29 @@ class KeyringController extends EventEmitter {
    *
    * Only supports 'Simple Key Pair'.
    *
-   * @param {string} type - The key pair type to check for.
-   * @param {Array<string>} newAccountArray - Array of new accounts.
-   * @returns {Promise<Array<string>>} The account, if no duplicate is found.
+   * @param type - The key pair type to check for.
+   * @param newAccountArray - Array of new accounts.
+   * @returns The account, if no duplicate is found.
    */
-  async checkForDuplicate(type, newAccountArray) {
+  async checkForDuplicate(
+    type: string,
```

Comment:
Do we have a list of valid keyring types somewhere that we can use here instead of accepting anything? If that's not possible right now, no problem, but it might be something we want to think about improving.

Reply:
I have defined this `enum`, we could use that one,

```ts
export enum KeyringType {
  HD = 'HD Key Tree',
  Simple = 'Simple Key Pair',
}
```

Reply:
But ultimately, there can be unknown keyrings type outside those